spec/requests/api/providers_spec.rb

```diff
@@ -744,6 +744,36 @@ def failed_auth_action(id)
       expect(response).to have_http_status(:ok)
       expect_results_to_match_hash("results", [failed_auth_action(p1.id), failed_auth_action(p2.id)])
     end
+
+    it "provider refresh are created with a task" do
+      api_basic_authorize collection_action_identifier(:providers, :refresh)
+
+      provider = FactoryGirl.create(:ext_management_system, sample_vmware.symbolize_keys.except(:type))
+      provider.update_authentication(:default => default_credentials.symbolize_keys)
+      allow_any_instance_of(provider.class).to receive_messages(:authentication_status_ok? => true)
```

Comment:
Although still problematic, WDYT about doing something like:

```ruby
provider.authentication_type(:default).update(:status => "Valid")
```

to avoid the use of `any_instance_of`?